### ISSUE TITLE
clang_delta assert: clang_delta: ../tools/clang/lib/AST/ExprConstant.cpp:12374 failed on i586

### ISSUE THREAD

**Issue by marxin** [csmith-project/creduce#212]
I see the following assert being triggered on `i586` and `armv7l`:

```
./clang_delta tests/reduce-array-dim/non-type-temp-arg.cpp --transformation=reduce-array-dim --counter=1
clang_delta: ../tools/clang/lib/AST/ExprConstant.cpp:12374: bool clang::Expr::EvaluateAsInt(clang::Expr::EvalResult &, const clang::ASTContext &, clang::Expr::SideEffectsKind, bool) const: Assertion `!isValueDependent() && "Expression evaluator can't be called on a dependent expression."' failed.
```

**Comment by chenyang78** [comment]
This was fixed. Thanks!

**Comment by marxin** [comment]
Can you please link a commit that fixed that with issues? It's then easier to find which issue was fixed by which commit. Thanks!

**Comment by chenyang78** [comment]
It was fixed in commit [34881e7](https://github.com/csmith-project/creduce/commit/34881e78f4ec4ed9f893da5179086c628c8f4f1d). Thanks.
